android/engine/src/main/java/org/smartregister/fhircore/engine/data/domain/util/DomainMapperWithDomainModelSource.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.data.domain.util
+
+/**
+ * Mapper util for transforming model of type [T] and one [DomainModelSource] to [DomainModel]. An
+ * example is transforming FHIR resource to an entity class
+ */
+interface DomainMapperWithDomainModelSource<T, DomainModelSource, DomainModel> {
```

Comment:
why do we need this?

Reply:
Used it for mapping if we have more then resource to.